<?xml version="1.0" encoding="utf-8"?>

<launch>

 <arg name="respawn_value" default="false" /> <!-- boolean: true, false -->
  
  <arg name="robot_name" default="ugv1" />
  <arg name="simulator" value="vrep" />
  
  <arg name="pcl_out" default="/volumetric_mapping/octomap_pcl" />

  <!-- octomap_manager from volumetric_mapping package -->
  <node name="volumetric_mapping" pkg="octomap_world" type="octomap_manager"  respawn="$(arg respawn_value)" >
    <param name = "tf_frame" value = "/map" />
    <param name = "robot_frame" value = "/$(arg robot_name)/base_link" />
    <param name = "resolution" value = "0.075" />
    <param name = "map_publish_frequency" value = "0.3" />
    <param name = "sensor_max_range" value = "20.0" />
    <!-- <param name = "Q" value = "" /> -->
    <!-- <param name = "octomap_file" value = "" /> -->
    <param name = "probability_hit" value = "0.75" />
    <!--param name = "probability_miss" value = "0.2" /-->
    <param name = "probability_miss" value = "0.45" />
    <param name = "threshold_min" value = "0.12" />
    <param name = "threshold_max" value = "0.97" />
    <param name = "threshold_occupancy" value = "0.5" />
    <param name = "filter_speckles" value = "false" />
    <!-- <param name = "visualize_min_z" value = "" /> -->
    <!-- <param name = "visualize_max_z" value = "" /> -->
    <!-- <param name = "full_image_width" value = "" /> -->
    <!-- <param name = "full_image_height" value = "" /> -->
    <!-- <param name = "treat_unknown_as_occupied" value = "true" /> -->
    <!-- <param name = "change_detection_enabled" value = "false" /> -->
    <param name = "dynamics_detection_enabled" value = "true" />

    <!-- Input -->
    <!--remap from = "pointcloud" to = "/laser_mapper/new_fixed_cloud" /-->
    <remap from = "pointcloud" to = "/volumetric_mapping/input_pcl" />
    
    <!-- Output -->
    <remap from = "/volumetric_mapping/octomap_pcl" to = "$(arg pcl_out)"/>
        
  </node>
    
</launch>
